MoreLinq/Lag.cs

```diff
@@ -94,5 +94,40 @@ public static partial class MoreEnumerable
                 }
             }
         }
+
+        /// <summary>
+        /// Produces a sequence of tuple containing a pair of elements separated by a negative offset.
+        /// </summary>
+        /// <remarks>
+        /// This operator evaluates in a deferred and streaming manner.<br/>
+        /// For elements prior to the lag offset, <c>default(T)</c> is used as the lagged value.<br/>
+        /// </remarks>
+        /// <typeparam name="TSource">The type of the elements of the source sequence</typeparam>
+        /// <param name="source">The sequence over which to evaluate lag</param>
+        /// <param name="offset">The offset (expressed as a positive number) by which to lag each value of the sequence</param>
+        /// <returns>A sequence of element of the sequence with its lagged pairing</returns>
+
+        public static IEnumerable<(TSource Current, TSource Lagged)> Lag<TSource>(this IEnumerable<TSource> source, int offset)
```

Comment:
Technically, the first element is the one that _lags_. I think the notion of _current_ doesn't make sense as part of a sequence or result, jus the cursor or iteration. I would think in terms of _item_. How about using `Item` and `OffsetItem` as names, as follows?

```suggestion
        public static IEnumerable<(TSource Item, TSource OffsetItem)> Lag<TSource>(this IEnumerable<TSource> source, int offset)
```

The names would be same for `Lag` and `Lead`, which is a fair parity.